### PR TITLE
feat: update to Gemini-1.5-pro SFT

### PR DESCRIPTION
**Pull request by dmoonat**
# Description

Supervised finetuning using gemini-1.5-pro

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: gemini/tuning/supervised_finetuning_using_gemini.ipynb [hunk 1]
```diff
@@ -419,7 +489,137 @@
...
+      "outputs": [],
+      "source": [
+        "def save_jsonlines(file, instances):\n",
+        "    \"\"\"\n",
+        "    Saves a list of json instances to a jsonlines file.\n",
+        "    \"\"\"\n",
+        "    with jsonlines.open(file, mode=\"w\") as writer:\n",
+        "        writer.write_all(instances)"
+      ]
+    },
```
Comment (holtskinner) [inline gemini/tuning/supervised_finetuning_using_gemini.ipynb:550]: For JSONL I/O I recommend using Pandas to directly download/output to GCS. I find it easier/cleaner to work with. (If it works for this particular transformation)

Example: https://github.com/GoogleCloudPlatform/generative-ai/pull/1138#discussion_r1767113650

**Review by holtskinner — APPROVED** [review]

**Comment by holtskinner** [comment]
Please add the spelling terms to the https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/actions/spelling/allow.txt (except maybe `jsonlines` unless you decide to keep that library)

File: gemini/tuning/supervised_finetuning_using_gemini.ipynb (hunk 1 above)
Comment (dmoonat) [reply gemini/tuning/supervised_finetuning_using_gemini.ipynb:550]: Thanks for the suggestion. 
`instances` is a list of json objects, and there is no use of dataframe that's why using `jsonlines`

Comment (dmoonat) [reply gemini/tuning/supervised_finetuning_using_gemini.ipynb:550]: For next time, will keep it in mind, thanks

**Pull request merged**